src/Relude/Bool/Guard.hs

```diff
@@ -78,6 +79,16 @@ guardM :: MonadPlus m => m Bool -> m ()
 guardM f = f >>= guard
 {-# INLINE guardM #-}
 
+-- | 'guarded'. Either lifts a value into an alternative context or gives a minimal value depending on a predicate.
+--
+-- @
+-- guarded even 3 :: [Int] -- []
+-- guarded even 2 :: [Int] -- [2]
+-- @
+guarded :: Alternative f => (a -> Bool) -> a -> f a
+guarded pred a = if pred a then pure a else empty
```

Comment:
The implementation is correct :+1: Let's just use `p` instead of `pred` as an idiomatic name for non-important polymorphic predicates in Haskell. Also, `pred` is a function in `Relude` that takes a previous number, so the implementation can be confusing...
```suggestion
guarded p a = if p a then pure a else empty
```